test/Integration.t.sol

```diff
@@ -0,0 +1,370 @@
+
+        // Check correct cap increase after cooldown
+        skip(24 hours);
+        capAutomator.execSupply(WBTC);
+        // initialSupply + suppliedBefore + newlySupplied + gap = 750 + 2_000 + 250 + 500 = 3_500
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_500);
+
+        vm.roll(block.number + 1);
+        vm.prank(user);
+        pool.withdraw(WBTC, 125e8, user);
+
+        // Check correct cap decrease
+        capAutomator.execSupply(WBTC);
+        // previousSupply + gap - justWithdrawn = 3_000 + 500 - 125 = 3_375
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_375);
+
+        vm.prank(user);
+        pool.withdraw(WBTC, 125e8, user);
+
+        // Check the cap is not changing in the same block
+        capAutomator.execSupply(WBTC);
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_375);
+
+        // Check correct cap decrease after block changes
+        vm.roll(block.number + 1);
+        capAutomator.execSupply(WBTC);
+        // previousSupply + gap - justWithdrawn = 2_875 + 500 - 125 = 3_250
+        assertEq(pool.getReserveData(WBTC).configuration.getSupplyCap(), 3_250);
+    }
+
+    function test_E2E_borrow_weth() public {
+        assertEq(ERC20(WETH).decimals(), 18);
+
+        DataTypes.ReserveData memory wethReserveData = pool.getReserveData(WETH);
+
+        // Confirm initial borrow cap
+        uint256 initialBorrowCap = wethReserveData.configuration.getBorrowCap();
+        assertEq(initialBorrowCap, 1_400_000);
+
+        // Confirm initial borrows
+        uint256 initialBorrows = currentBorrows(wethReserveData);
+        assertEq(initialBorrows, 126_520);
+
+        vm.prank(SPARK_PROXY);
+        capAutomator.setBorrowCapConfig({
+            asset:            WETH,
+            max:              2_000_000,
+            gap:              100_000,
+            increaseCooldown: 12 hours
+        });
+
+        vm.startPrank(user);
+
+        deal(WBTC, user, USERS_STASH);
+        ERC20(WBTC).approve(POOL, USERS_STASH);
+
+        pool.supply(WBTC, 2_000e8, user, 0);
+
+        vm.stopPrank();
+
+        // Check correct cap decrease
+        capAutomator.execBorrow(WETH);
+        // totalDebt + gap = 126_520 + 100_000 = 226_520
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 226_520);
+
+        vm.prank(user);
+        pool.borrow(WETH, 480e18, 2 /* variable rate mode */, 0, user);
+
+        // Check that another cap change is not possible in the same block
+        capAutomator.execBorrow(WETH);
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 226_520);
+
+        // Check correct cap increase in the new block
+        vm.roll(block.number + 1);
+        capAutomator.execBorrow(WETH);
+        // totalDebt + gap = initialBorrows + newlyBorrowed + gap = 126_520 + 480 + 100_000 = 227_000
+        assertEq(pool.getReserveData(WETH).configuration.getBorrowCap(), 227_000);
```

Comment:
I didn't use exactly the same flow/order, because the initial state of the market is different and the more natural first action is different for each market. Made sure though that both tests test the same functionalities.